### ISSUE TITLE
Type mismatch with hooks

### ISSUE THREAD

**Issue by Aeonax** [clj-kondo/clj-kondo#2943]
# type-mismatch false positive
Empty "received" type, no line/col on calls rewritten by :analyze-call hooks

## version
2026.08.03 (reproduced with the `cljkondo/clj-kondo:latest` docker image, linux).
Not reproducible on 2026.05.25 (macos binary) — same setup lints clean.

## problem

Since 2026.08.03, when a call is rewritten by an `:analyze-call` hook and the called
function has inferred parameter types (e.g. a param is passed to `get` in the fn body),
the `:type-mismatch` linter reports an error with an **empty "received" type** and
**no line/col**:

```
src/repro/core.clj::: error: Expected: string or array or nil or set or ILookup or map, received: .
```

Returning the original node unchanged from the hook (`{:node node}`) does not trigger
it; returning any newly constructed node does, even when the new node reuses the
original argument nodes untouched.

## repro

`.clj-kondo/config.edn`:

```clojure
{:hooks {:analyze-call {repro.core/dispatch hooks.repro/dispatch}}}
```

`.clj-kondo/hooks/repro.clj`:

```clojure
(ns hooks.repro
  (:require [clj-kondo.hooks-api :as api]))

(defn dispatch [{:keys [node]}]
  ;; rewrite (dispatch <args>) => (vector <args>), reusing the original arg nodes
  (let [[_ & args] (:children node)]
    {:node (api/list-node (list* (api/token-node 'vector) args))}))
```

`src/repro/core.clj`:

```clojure
(ns repro.core)

(defn dispatch [event-name client]
  (get client :k))

(defn use-it []
  (dispatch :event {:a 1}))
```

`clj-kondo --lint src`

```
src/repro/core.clj::: error: Expected: string or array or nil or set or ILookup or map, received: .
src/repro/core.clj:3:17: warning: unused binding event-name
linting took 19ms, errors: 1, warnings: 1
```

## expected behavior

No type-mismatch error

## additional observations

- An "identity hook" (`(defn dispatch [{:keys [node]}] {:node node})`) → no error.
- The shape of the rewrite does not matter: wrapping the untouched original node in
  `(do ...)` or generating several calls reproduces it as well.
- The actual argument type is ignored: passing a literal map (valid for `get`) still errors.

**Comment by Aeonax** [comment]
Ow, I didn't notice version 2026.07.24-_- It looks more valid for this issue😅🙈
